### PR TITLE
Tighten type information for `from_numpy`

### PR DESCRIPTION
**Pull request by inducer**
This fixes Mypy failures that have recently cropped up, e.g. https://github.com/inducer/arraycontext/runs/4751297347?check_suite_focus=true

cc @matthiasdiener

### CODE REVIEW

**Comment by inducer** [comment]
FWIW, grudge downstream CI is still broken: https://github.com/inducer/grudge/issues/202

I'll force-merge this if all but the grudge downstream passes.

File: arraycontext/container/traversal.py [hunk 1]
```diff
@@ -732,13 +733,16 @@ def _unflatten(template_subary: ArrayOrContainerT) -> ArrayOrContainerT:
...
+def from_numpy(
+        ary: Union[np.ndarray, np.generic, Number],
+        actx: ArrayContext) -> ArrayOrContainerT:
     """Convert all :mod:`numpy` arrays in the :class:`~arraycontext.ArrayContainer`
     to the base array type of :class:`~arraycontext.ArrayContext`.
 
     The conversion is done using :meth:`arraycontext.ArrayContext.from_numpy`.
     """
-    def _from_numpy_with_check(subary: Any) -> Any:
+    def _from_numpy_with_check(subary: Union[np.ndarray, np.generic, Number]) \
```
Comment (alexfikl) [inline arraycontext/container/traversal.py:744]: Last time I checked `mypy` didn't like `Number` in the type hints
https://github.com/python/mypy/issues/3186
Doesn't seem like there's a good workaround besides making another union of all the number types we care about :\

Comment (inducer) [reply arraycontext/container/traversal.py:744]: Thanks for taking a look! After briefly considering [this](https://posita.github.io/numerary/0.3/), I decided to go with the lower-fidelity :hankey: approach instead.

https://github.com/inducer/arraycontext/compare/7c2e18355cd0b2ee9861aa19d63bd508ecdc8ea0..88f88a6a774a3bf8bfb388925bf743d89e558fad

Comment (alexfikl) [reply arraycontext/container/traversal.py:744]: Yeah, that `_ScalarLike` looks good to me!

It seems to be missing the changes in `traversal.py`?

Comment (inducer) [reply arraycontext/container/traversal.py:744]: Whoops. Thanks for catching that! #140

**Pull request merged**